src/test/java/com/itkonboarding/airport_gate/services/GateServiceImpTest.java

```diff
@@ -0,0 +1,261 @@
+        Integer id = null;
+        var gate = mock(Gate.class);
+
+        gateServiceImp.create(id, gate);
+
+        assertAll(
+                () -> verify(gate).setStatus(AVAILABLE),
+                () -> verify(gateRepository).save(gate),
+                () -> verifyNoInteractions(airportServiceImp)
+        );
+    }
+
+    @Test
+    void create() {
+        var id = new Random().nextInt();
+        var gate = mock(Gate.class);
+        var airport = mock(Airport.class);
+
+        when(airportServiceImp.findById(id)).thenReturn(of(airport));
+
+        gateServiceImp.create(id, gate);
+
+        assertAll(
+                () -> verify(gate).setAirport(airport),
+                () -> verify(gate).setStatus(AVAILABLE),
+                () -> verify(gateRepository).save(gate)
+        );
+    }
+
+    @Test
+    void update_noGateFoundException() {
+        var gateId = new Random().nextInt();
+        var airportId = new Random().nextInt();
+        var gate = mock(Gate.class);
+
+        when(gateRepository.findById(gateId)).thenReturn(empty());
+
+        assertAll(
+                () -> assertThatExceptionOfType(ResourceNotFoundException.class)
+                        .isThrownBy(() -> gateServiceImp.update(gateId, airportId, gate))
+                        .withMessage(GATE_NOT_FOUND),
+                () -> verifyNoInteractions(gate),
+                () -> verifyNoMoreInteractions(gateRepository)
+        );
+    }
+
+    @Test
+    void update_noAirportFoundException() {
+        var gateId = new Random().nextInt();
+        var airportId = new Random().nextInt();
+        var gate = mock(Gate.class);
+
+        when(airportServiceImp.findById(airportId)).thenReturn(empty());
+        when(gateRepository.findById(gateId)).thenReturn(of(gate));
+
+        assertAll(
+                () -> assertThatExceptionOfType(ResourceNotFoundException.class)
+                        .isThrownBy(() -> gateServiceImp.update(gateId, airportId, gate))
+                        .withMessage(AIRPORT_NOT_FOUND),
+                () -> verifyNoMoreInteractions(gateRepository)
+        );
+    }
+
+    @Test
+    void update() {
+        var gateId = new Random().nextInt();
+        var airportId = new Random().nextInt();
+        var gate = mock(Gate.class);
+        var updatedGate = mock(Gate.class);
+        var gateName = make();
+        var airport = mock(Airport.class);
+
+        when(gate.getGateName()).thenReturn(gateName);
+        when(airportServiceImp.findById(airportId)).thenReturn(of(airport));
+        when(gateRepository.findById(gateId)).thenReturn(of(updatedGate));
+
+        gateServiceImp.update(gateId, airportId, gate);
+
+        assertAll(
+                () -> verify(updatedGate).setGateName(gateName),
+                () -> verify(updatedGate).setAirport(airport),
+                () -> verify(gateRepository).save(updatedGate)
+        );
+    }
+
+    @Test
+    void update_withNull() {
```

Comment:
This method is not necessary since you already tested  that gate is null in update_noAirportFoundException method